src/openforms/js/components/admin/form_design/form-creation-form.js

```diff
@@ -1267,8 +1270,20 @@ const FormCreationForm = ({formUuid, formUrl, formHistoryUrl}) => {
       {Object.keys(state.errors).length ? (
         <div className="fetch-error">
           <FormattedMessage
-            defaultMessage="The form is invalid. Please correct the errors below."
-            description="Generic error message"
+            defaultMessage="Something went unexpectedly wrong."
+            description="Generic error message title"
+          />
+          <br />
+          <FormattedMessage
+            defaultMessage="Contact your administrator or see the {link} for possible/more details."
+            description="Generic error message requests logs link"
+            values={{
+              link: (
+                <a href={adminRequestsLog} target="_blank" rel="noopener noreferrer">
+                  requests logs
+                </a>
+              ),
+            }}
           />
```

Comment:
I think it's better to keep this as a single message, and make sure you can translate the label of the link too:

```jsx
<FormattedMessage
  description="Generic admin error message"
  defaultMessage={`Sorry! Something unexpected went wrong.<br></br>Contact your 
  technical administrator to investigate, or perhaps more information is 
  available in the <link>outgoing request logs</link>.`}
  values={{
    br: () => <br/>,
    link: (chunks) => <a href="/admin/..." target="_blank">{chunks}</a>,
  }}
/>
```